docs/extending/developing-code/code-overview/evaluation.rst

```diff
@@ -36,37 +38,35 @@ that, the method handle ``b`` needs to be computed. And this comes
 from a class heirarchy.
 
 Mathics and WL are not Object Oriented, so there is no such
-class-hierarchy lookup.  However, as mentioned above, pattern matching
+class-hierarchy lookup.  Instead, as mentioned above, pattern matching
 is used to decide which method of the object to call.
 
 Function Name to Python method lookup
--------------------------------------
+=====================================
 
 .. index:: Symbol, Predefined, Builtin, Expression
 
 When an ``Expression`` has not been rewritten, the entire function
 invocation in Mathics comes from the first leaf (or ``Head[]``) which
 should be a ``Symbol``. In Python this will be a class some sort, such
```

Comment:
Head[] could also be an expression, like in `Derivative[1,2][f][x,y]`. In this case,
`Head[Derivative[1,2][f][x,y]]` is `Derivative[1,2][f]` and
`Head[Head[Derivative[1,2][f][x,y]]]` is `Derivative[1,2]`.
Of course, the typical expressions have Symbols as heads.

Reply:
Do I have this correct that if the head is an expression then after it is evaluated the result should be a function which is then applied to the remaining leaves?